File: database/models/Drug.js
import mongoose from 'mongoose';

const drugSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true
  },
  category: {
    type: String,
    required: true,
    trim: true
  },
  type: {
    type: String,
    required: true,
    trim: true
  },
  activeIngredient: {
    type: String,
    required: true,
    trim: true
  },
  manufacturer: {
    type: String,
    required: true,
    trim: true
  },
  uses: {
    type: String,
    required: true,
    trim: true
  },
  dosage: {
    type: String,
    required: true,
    trim: true
  },
  contraindications: {
    type: String,
    required: true,
    trim: true
  },
  sideEffects: {
    type: String,
    required: true,
    trim: true
  },
  createdAt: {
    type: Date,
    default: Date.now
  }
});

// Tạo index để tìm kiếm nhanh hơn
drugSchema.index({ name: 'text', activeIngredient: 'text', uses: 'text' });
drugSchema.index({ category: 1, type: 1 });

const Drug = mongoose.model('Drug', drugSchema);

export default Drug; 

File: flutterApp/flutter_application_1/lib/models/drug.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'drug.freezed.dart';
part 'drug.g.dart';

@freezed
class Drug with _$Drug {
  const factory Drug({
    String? id,
    required String name,
    String? category,
    String? type,
    String? activeIngredient,
    String? manufacturer,
    String? uses,
    String? dosage,
    String? contraindications,
    String? sideEffects,
  }) = _Drug;

  factory Drug.fromJson(Map<String, dynamic> json) => _$DrugFromJson(json);
} 